agent/agentserver/server_test.go

```diff
@@ -276,13 +277,82 @@ func TestReadinessCheckHandler(t *testing.T) {
 	}
 }
 
+func TestReadinessCheckHandlerCache(t *testing.T) {
+	for _, tc := range []struct {
+		desc              string
+		firstCallErr      error
+		readinessCacheTTL time.Duration
+		waitInvalidation  bool
+	}{
+		{
+			desc:              "call 1 succeeds, so second call automatically succeeds",
+			firstCallErr:      nil,
+			readinessCacheTTL: 10 * time.Minute,
+			waitInvalidation:  false,
+		},
+		{
+			desc:              "call 1 fails, so second call performs checks",
+			firstCallErr:      errors.New("test-error"),
+			readinessCacheTTL: 10 * time.Minute,
+			waitInvalidation:  false,
+		},
+		{
+			desc:              "call 1 succeeds, but cache becomes invalid, so second call performs checks",
+			firstCallErr:      nil,
+			readinessCacheTTL: 10 * time.Minute,
+			waitInvalidation:  true,
+		},
+		{
+			desc:              "call 1 succeeds, but caching is disabled, so second call performs checks",
+			firstCallErr:      nil,
+			readinessCacheTTL: 0,
+			waitInvalidation:  false,
+		},
+	} {
+		t.Run(tc.desc, func(t *testing.T) {
+			require := require.New(t)
+			mocks, cleanup := newServerMocks(t)
+			defer cleanup()
+
+			mocks.sched.EXPECT().Probe().Return(tc.firstCallErr).Times(1)
+			mocks.tags.EXPECT().CheckReadiness().Return(tc.firstCallErr).Times(1)
+			mocks.ac.EXPECT().CheckReadiness().Return(tc.firstCallErr).Times(1)
+			if tc.firstCallErr != nil || tc.waitInvalidation || tc.readinessCacheTTL == 0 {
+				mocks.sched.EXPECT().Probe().Return(nil).Times(1)
+				mocks.tags.EXPECT().CheckReadiness().Return(nil).Times(1)
+				mocks.ac.EXPECT().CheckReadiness().Return(nil).Times(1)
+			}
+
+			s, addr := mocks.startServer(Config{readinessCacheTTL: tc.readinessCacheTTL})
```

Comment:
optional: sometimes when testing retry or caching logic where you need multiple calls you can parameterize the calls you want and what result you expect from them. It can just be a slice of structs with all the params/expected result you want for that call. Then you can loop through and make a call for each call entry in the test body.

Reply:
Done.